app/src/main/java/fr/neamar/kiss/dataprovider/SearchProvider.java

```diff
@@ -10,13 +11,17 @@
 import java.util.Map;
 import java.util.Set;
 import java.util.TreeSet;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 import fr.neamar.kiss.loader.LoadSearchPojos;
 import fr.neamar.kiss.pojo.Pojo;
 import fr.neamar.kiss.pojo.SearchPojo;
 
 public class SearchProvider extends Provider<SearchPojo> {
     private SharedPreferences prefs;
+    public static final String URL_REGEX = "^((https?|ftp)://|(www|ftp)\\.)?[a-z0-9-]+(\\.[a-z0-9-]+)+([/?].*)?$";
```

Comment:
@EnricoMonese you probably need to escape the "\x00" to read "\\x00". The first reverse slash escape it for Java, and it's then sent to the regexp.

I might be wrong though, but I believe that's why it's failing (that and all the other \ on the regexp)